Docs: update connector test documentation (#44126)
airbytehq/airbyte: docs/connector-development/testing-connectors/README.md

Multiple tests suites compose the Airbyte connector testing pyramid

## Tests run by our CI pipeline

> - [Connectors QA checks](https://docs.airbyte.com/contributing-to-airbyte/resources/qa-checks): Static assets checks that validate that the connector is correctly packaged to be successfully released to production.

Reviewer: ```suggestion
- [Connectors QA checks](https://docs.airbyte.com/contributing-to-airbyte/resources/qa-checks): Static asset checks that validate that a connector is correctly packaged to be successfully released to production.
```

Reviewer: Let's capitalize the whole test: Connector QA Checks

> - Unit tests: Connector-specific tests written by the connector developer and which don’t require access to the source/destination.

Reviewer: ```suggestion
- Unit tests: Connector-specific tests written by the connector developer which don’t require access to the source/destination.
```

Reviewer: Let's capitalize the whole test: Unit Tests

> - Integration tests: Connector-specific tests written by the connector developer which may require access to the source/destination.

Reviewer: ```suggestion
- Integration tests: Connector-specific tests written by the connector developer which _may_ require access to the source/destination.
```

Reviewer: Let's capitalize the whole test: Integration Tests

> - [Connector Acceptance tests](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference/): Connector-agnostic tests that verify that the connector adheres to the Airbyte protocol. Credentials to a source/destination sandbox account are required.

Reviewer: ```suggestion
- [Connector Acceptance tests](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference/): Connector-agnostic tests that verify that a connector adheres to the Airbyte protocol. Credentials to a source/destination sandbox account are **required**.
```

Reviewer: can you add a link to the protocol doc?

Reviewer: Let's capitalize the whole test: Connector Acceptance Tests

> - [Regression tests](https://github.com/airbytehq/airbyte/tree/master/airbyte-ci/connectors/live-tests): Connector-agnostic tests that verify that the behavior of the connector hasn’t changed unexpectedly between connector versions. A sandbox cloud connection is required. Currently only available for API source connectors.

Reviewer: ```suggestion
- [Regression Tests](https://github.com/airbytehq/airbyte/tree/master/airbyte-ci/connectors/live-tests): Connector-agnostic tests that verify that the behavior of a connector hasn’t changed unexpectedly between connector versions. A sandbox cloud connection is required. Currently only available for API source connectors.
```

airbyte-ci connectors --name=<connector_name> test
```

> This will run all the tests that are available for the connector. This can include all of the tests listed above, if we have the appropriate credentials; at a minimum it will include the static asset checks and any tests that exist in a connector's `unit_tests` and `integration_tests` directories.

Reviewer: ```suggestion
CI will run all the tests that are available for a connector. This can include all of the tests listed above, if we have the appropriate credentials. At a minimum, it will include the static asset checks and any tests that exist in a connector's `unit_tests` and `integration_tests` directories.
```

Reviewer: do we have any more information on static asset checks?

Author: This was just referring to the Connector QA checks. Updated.
